tests/sync/sync_manager.cpp

```diff
@@ -0,0 +1,180 @@
+////////////////////////////////////////////////////////////////////////////
+//
+// Copyright 2016 Realm Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+////////////////////////////////////////////////////////////////////////////
+
+#include "sync_test_utils.hpp"
+
+#include "sync/sync_manager.hpp"
+#include "sync/sync_user.hpp"
+#include <realm/util/logger.hpp>
+
+using namespace realm;
+using namespace realm::util;
+using File = realm::util::File;
+
+static const std::string base_path = "/tmp/realm_objectstore_sync_manager/";
+
+namespace {
+
+bool validate_user_in_vector(std::vector<std::shared_ptr<SyncUser>> vector,
+						     const std::string& identity,
+						     const std::string& url,
+						     const std::string& token) {
+	for (auto& user : vector) {
+		if (user->identity() == identity && user->server_url() == url && user->refresh_token() == token) {
+			return true;
+		}
+	}
+	return false;
+}
+
+}
+
+TEST_CASE("sync_manager: basic property APIs") {
+	reset_test_directory(base_path);
+	SyncManager::shared().configure_file_system(base_path, SyncManager::MetadataMode::NoMetadata);
+
+	SECTION("should work for log level") {
+		SyncManager::shared().set_log_level(util::Logger::Level::info);
+		REQUIRE(SyncManager::shared().log_level() == util::Logger::Level::info);
+		SyncManager::shared().set_log_level(util::Logger::Level::error);
+		REQUIRE(SyncManager::shared().log_level() == util::Logger::Level::error);
+	}
+
+	SECTION("should work for 'should reconnect immediately'") {
+		SyncManager::shared().set_client_should_reconnect_immediately(true);
+		REQUIRE(SyncManager::shared().client_should_reconnect_immediately());
+		SyncManager::shared().set_client_should_reconnect_immediately(false);
+		REQUIRE(!SyncManager::shared().client_should_reconnect_immediately());
+	}
+
+	SECTION("should work for 'should validate SSL'") {
+		SyncManager::shared().set_client_should_validate_ssl(true);
+		REQUIRE(SyncManager::shared().client_should_validate_ssl());
+		SyncManager::shared().set_client_should_validate_ssl(false);
+		REQUIRE(!SyncManager::shared().client_should_validate_ssl());
+	}
+
+	SyncManager::shared().reset_for_testing();
```

Comment:
The call to `reset_for_testing()` should probably be always done as a scope_exit call set up immediately after the `TEST_CASE` line. Calling it manually at the end of the test case means that it won't get called if any of the tests fail (presumably resulting in all of the subsequent tests failing and adding  a lot of noise to the report), and calling it within each `SECTION` as is done in some of the tests is redundant (only a single leaf SECTION is run per execution of the test case).